src/amplify-migration/amplifyMigrationScript.ts

```diff
@@ -0,0 +1,386 @@
+const simpleGit = require("simple-git");
+const fs = require("fs");
+const glob = require("glob");
+const path = require("path");
+const csv = require("csv-parser");
+const os = require("os");
+const axios = require("axios");
+const { Octokit } = require("@octokit/core");
+require("dotenv").config();
+
+const { JSDOM } = require("jsdom");
+const {
+  createAmplifyBranches,
+  startReleaseJob,
+  createAmplifyApp,
+  readBuildSpec,
+} = require("./amplifyUtils");
+const BRANCH_NAME = "chore-amplify-migration-change-permalinks";
+const ORGANIZATION_NAME = "isomerpages";
+const PERMALINK_REGEX = /^permalink: /m;
+
+interface AmplifyAppInfo {
+  appId: string;
+  repoName: string;
+  name: string;
+  repoPath: string;
+}
+
+function readCsvFile(): Promise<[string, string][]> {
+  // read file list-of-repos.csv
+  const filePath = path.join(__dirname, "list-of-repos.csv");
```

Comment:
any reason why we're not letting the users specify the csv directly using the cli?

if this is only to be used by you, it's ok but otherwise is the script itself idempotent? (seems like hor so should be ok)

Reply:
Sure, updated here: 66708a6
 
> otherwise is the script itself idempotent?

hmm no leh, got quite a number of side effects. (eg. creation of amplify apps, sql commands written to db). The goal of this script was never idempotency tho. 

to be truly idempotent, we would need to talk to the db, which can be tackled in a seperate PR ya